src/main/ruby/core/enumerable.rb

```diff
@@ -716,8 +716,17 @@ def max_n(n, &block)
   end
   private :max_n
 
-  def max_by
+  def max_by_n(n, &block)
+    n = Rubinius::Type.rb_num2long(n)
+    raise ArgumentError, "negative size #{n}" if n < 0
+    return [] if n == 0
+    self.sort_by(&block).last(n).reverse
+  end
+  private :max_by_n
```

Comment:
I personally prefer `private def` as it clarifies it's a helper method from the start.

Reply:
Okay, updated to `private def`